include/onnxruntime/core/framework/ortdevice.h

```diff
@@ -26,6 +26,7 @@ struct OrtDevice {
     static const MemoryType HIP_PINNED = 2;
     static const MemoryType CANN_PINNED = 3;
     static const MemoryType QNN_HTP_SHARED = 4;
+    static const MemoryType CPU_ALIGNED_4K = 5;
```

Comment:
This doesn't really feel like a new memory type. Is there another way to do this? Would an optional alignment field be clearer? Allocator could honor that if set.

Reply:
It does feel this way, but it is another memory type. The device type and memory type together are really the keys to request the specific allocator. The provider gives OrtDevice to the allocation planner, it records it into the plan, then the execution frame uses OrtDevice as a key to the map of the allocators.

Reply:
It is not only about alignment. Qnn may return a device specific allocator that is CPU accessible.